numpy/lib/function_base.py

```diff
@@ -356,7 +356,7 @@ def jhat(nbins):
         p_k = np.histogram(x, bins=nbins)[0] / n
         return (2 - (n + 1) * p_k.dot(p_k)) / hh
 
-    nbins = min(range(1, max(n // 4, 2)), key=jhat)
+    nbins = min(range(1, max(100, int(np.sqrt(n)))), key=jhat)
```

Comment:
Are you sure you didn't mean `min`? What are you trying to achieve here? Clearly, a data set of length `1` should not try `100` bins